AnyType.cpp

```diff
@@ -0,0 +1,156 @@
+#include "AnyType.h"
+
+
+AnyType::AnyType()
+{
+    data.double_var = 0;
+    type_hash = 0;
+}
+
+AnyType::AnyType(bool val)
+{
+    data.bool_var = val;
+    type_hash = typeid(bool).hash_code();
+}
+
+AnyType::AnyType(char val)
+{
+    data.char_var = val;
+    type_hash = typeid(char).hash_code();
+}
+
+AnyType::AnyType(int val)
+{
+    data.int_var = val;
+    type_hash = typeid(int).hash_code();
+}
+
+AnyType::AnyType(float val)
+{
+    data.float_var = val;
+    type_hash = typeid(float).hash_code();
+}
+
+AnyType::AnyType(double val)
+{
+    data.double_var = val;
+    type_hash = typeid(double).hash_code();
+}
+
+AnyType::AnyType(const AnyType& copy)
+{
+    data = copy.data;
+    type_hash = copy.type_hash;
+}
+
+AnyType::AnyType(AnyType&& copy)
+{
+    data = copy.data;
```

Comment:
Have you considered to use std::move for move constructor

Reply:
Yes, but my class consists of built-in types, so std::move is an equivalent to copying.